Source/WebCore/rendering/AutoTableLayout.cpp

```diff
@@ -249,6 +249,13 @@ void AutoTableLayout::computeIntrinsicLogicalWidths(LayoutUnit& minWidth, Layout
             maxWidth = m_scaledWidthFromPercentColumns;
     }
 
+    if (!scaleColumnsForSelf && m_layoutStruct.isEmpty()) {
```

Comment:
so TableIntrinsics::ForLayout is legacy behavior and TableIntrinsics::ForKeyword is proper behavior.